### PR TITLE
fix(client): installing auth plugin to services fails

### PR DESCRIPTION
**Pull request by Meyazhagan**
Close: #6661 

### Summary
🔧🔄🚧

Fixed a bug in the plugins catalog that caused incorrect plugin toggling and a potential error when loading plugin data. Changed the `pluginId` property of the `data` object in `PluginsCatalog.tsx` and made the `plugin` object optional in the `href` attribute of the anchor element in `PluginsCatalogItem.tsx`.

> _`pluginId` is key_
> _to toggle the right plugin_
> _avoid errors too_

### Walkthrough
*  Use `pluginId` instead of `id` to identify plugins in `PluginsCatalog` ([link](https://github.com/amplication/amplication/pull/6722/files?diff=unified&w=0#diff-c04b321b9fbe9466d4437357cde44fb0d2d37ff0f142837eadadc332969f99cbL185-R185))
*  Make `plugin` object optional in `href` attribute of anchor element in `PluginsCatalogItem` to avoid error when plugin data is not loaded ([link](https://github.com/amplication/amplication/pull/6722/files?diff=unified&w=0#diff-069de3b168a7ec6810b1453bcb08a63346d53aff23003da53f0ea774bd4324e1L140-R140))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Plugins/PluginsCatalogItem.tsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ function PluginsCatalogItem({
       <div className={`${CLASS_NAME}__row `}>
         <span className="spacer" />
         <span className={`${CLASS_NAME}__repo`}>
-          <a href={plugin.github} target="github_plugin">
+          <a href={plugin?.github} target="github_plugin">
```
Comment (morhag90) [inline packages/amplication-client/src/Plugins/PluginsCatalogItem.tsx:140]: @Meyazhagan Why did you add the ? mark?

Comment (Meyazhagan) [reply packages/amplication-client/src/Plugins/PluginsCatalogItem.tsx:140]: Plugin is `undefined` for some cases (like while loading or no plugin definition) so I handled for those cases

**Comment by Meyazhagan** [comment]
@morhag90
